### ISSUE TITLE
Proposal: Drop integrated tidb support

### ISSUE THREAD

**Issue by sapk** [go-gitea/gitea#3990]
TiDB since v2 seems to drop the integrated database model and shift to provide a scalable/distributed/mysql compatible independant database server.
So I propose to drop support for the old version that seems to not be use and we haven't provided test or binary for it. We should still be compatible since TiDB is compatible with mysql.

I might be misleading so this PR is to find out if someone use it or if everyone is ok with this proposal.

**Comment by sapk** [comment]
~~~And it seems that gitea doesn't build with tidb tags :~~~
I had some changes in /vendor kept from when trying to update to last v1 version that was failing.

**Comment by lunny** [comment]
Since we could use xgo to cross-compile sqlite. I think remove embed tidb support is acceptable.

**Comment by sapk** [comment]
PR ready as this is easy (#3993) but before I think we should wait if anyone complain.

**Comment by techknowlogick** [comment]
What if we announce that we are going to do this in the 1.5.0 release blog post, and then merge it into master right after 1.5.0 is released that way it will be in 1.6.0
